src/goto-instrument/code_contracts.cpp

```diff
@@ -370,25 +402,74 @@ void code_contractst::add_contract_check(
   dest.destructive_insert(dest.instructions.begin(), check);
 }
 
-void code_contractst::operator()()
+bool code_contractst::replace_calls(
+  const std::list<std::string> &funs_to_replace)
 {
-  Forall_goto_functions(it, goto_functions)
-    code_contracts(it->second);
-
-  goto_functionst::function_mapt::iterator i_it=
-    goto_functions.function_map.find(INITIALIZE_FUNCTION);
-  assert(i_it!=goto_functions.function_map.end());
+  bool fail = false;
+  Forall_goto_functions(fit, goto_functions)
```

Comment:
Do we still need to use the Forall_ macros, or do range-based loops now work? I had a feeling the did after some of the various cleanups that have gone into CBMC.

Reply:
Regular `for(auto &goto_function : goto_functions.function_map)` should work here.